src/components/Blog/BlogArticlesBoard/BlogArticlesBoard.tsx

```diff
@@ -37,76 +42,171 @@ export const BlogArticlesBoard = ({
         value: el.id,
-        onClick: handleTagsClick(el.id, el.attributes.Title),
+        onClick: handleTagsClick(index, el.attributes.Title),
         // disabled: false,
       };
     });
     // merge default + tags
-    output?.unshift(defaultFilter);
+    // output?.unshift(defaultFilter);
     return output;
-  }, [categoryId, catLabel, handleTagsClick, isDesktop, t, tags, theme]);
+  }, [handleTagsClick, tags?.data]);
+
+  const handleClick = (event: { target: HTMLElement }) => {
+    const classList = event.target.classList;
+
+    if (openDropdown) {
+      !classList?.contains(ariaLabel) && setOpenDropdown(false);
+    }
+  };
 
   return (
-    <BlogArticlesBoardContainer id="see-all">
+    <BlogArticlesBoardContainer
+      id="see-all"
+      onClick={(event: React.MouseEvent<HTMLDivElement>) =>
+        handleClick({ target: event.target as HTMLElement })
+      }
+    >
       <BlogArticlesBoardTitle variant="headerMedium">
         {t('blog.categories')}
       </BlogArticlesBoardTitle>
       {filteredTags ? (
         <BlogArticlesBoardTabs
+          ariaLabel={ariaLabel}
           openDropdown={openDropdown}
           filteredTags={filteredTags}
-          categoryId={categoryId}
+          tabId={tabId}
         />
       ) : (
         <Skeleton sx={{ width: '100%', height: 68 }} />
       )}
-      <Fade in={!!data} timeout={600}>
-        <></>
-        {/* todo: enable */}
-        {/* <ArticlesGrid container>
-          {!!data ? (
-            Array.from({ length: pageSize }).map((_, index) => (
-              <BlogArticleCardSkeleton
-                key={`blog-article-card-skeleton-${categoryId}-${index}`}
-              />
-            ))
-          ) : data?.length > 0 ? (
-            data?.map((article: BlogArticleData, index: number) => (
+      {/* <Fade in={!!data} timeout={600}> */}
+      {!data ? (
+        Array.from({ length: tags.meta.pagination.pageSize }).map(
+          (_, index) => (
+            <BlogArticleCardSkeleton
+              key={`blog-article-card-skeleton-${tabId}-${index}`}
+            />
+          ),
+        )
+      ) : tags.data?.length > 0 ? (
+        tags.data?.map((tag, tagIndex: number) => {
+          const pagination = {
+            page: 0,
+            pageSize: pageSize,
+            pageCount: Math.ceil(
+              tag.attributes.blog_articles.data.length / pageSize,
+            ),
+            total: tag.attributes.blog_articles.data.length,
+          };
+          return (
+            <CategoryTabPanel
+              value={tabId || 0}
+              index={tagIndex}
+              tags={tags}
+              pagination={pagination}
+              ariaLabel={ariaLabel}
+              data={tag.attributes.blog_articles.data}
+            />
+          );
+        })
+      ) : (
+        <p>No Content</p> //todo: find better option
+      )}
+      {/* </Fade> */}
+    </BlogArticlesBoardContainer>
+  );
+};
+
+interface CategoryTabPanelProps {
```

Comment:
Please split it out, there's 2 components into the same file (+ interfaces)

Reply:
Yes, done